src/main/java/htsjdk/samtools/reference/AbstractIndexedFastaSequenceFile.java

```diff
@@ -146,6 +147,55 @@ protected static void sanityCheckDictionaryAgainstIndex(final String fastaFile,
         }
     }
 
+    /** Do some basic checking to make sure the fasta and the index match.
+     * <p>
+     * checks that the length of the fasta file is at least as long as the index proclaims
+     * and that beyond the last position references in the index there is only one line followed by whitespaces
+     *
+     * @param fastaFile Path to fasta file
+     * @param fastaSequenceIndexes index file to check against the fasta file.
+     *
+     * @throws IOException in case of io-error when reading fastaFile
+     */
+    public void sanityCheckFastaAgainstIndex(final Path fastaFile,
+                                             final FastaSequenceIndex fastaSequenceIndexes) throws IOException {
+
+        final FastaSequenceIndexEntry lastSequenceIndex = fastaSequenceIndexes.getIndexEntry(fastaSequenceIndexes.getLastSequence());
+
+        final long lastSequenceLength = lastSequenceIndex.getSize();
+        final long lastSequenceStart = lastSequenceIndex.getLocation();
+        final long lastSequenceEnd = lastSequenceStart + lastSequenceIndex.getOffset(lastSequenceLength);
+
+        final long fastaLength = Files.size(fastaFile);
+        //Question: should we worry about files with lots of whitespace in their end?
```

Comment:
I wouldn't worry about "lots of whitespace" at the end.  I think that is still technically valid fasta.